statedb/treeupdate.go

```diff
@@ -140,16 +147,78 @@ func (u *TreeUpdate) subTree(cfg *treeConfig) (treeUpdate *TreeUpdate, err error
 	return treeUpdate, nil
 }
 
-// SubTreeSingle returns a TreeUpdate of a singleton SubTree whose root is stored
-// in the leaf with `cfg.Key()`, and is parametrized by `cfg`.
-func (u *TreeUpdate) SubTreeSingle(cfg *SubTreeSingleConfig) (*TreeUpdate, error) {
-	return u.subTree(cfg.treeConfig())
+// DeepSubTree allows opening a nested subTree by passing the list of tree
+// configurations.
+func (u *TreeUpdate) DeepSubTree(cfgs []*TreeConfig) (treeUpdate *TreeUpdate, err error) {
```

Comment:
would be nicer to use if it was like `cfgs ...*TreeConfig`, then you can call it like `DeepSubTree(cfg1)` o `DeepSubTree(cfg1, cfg2)` without having to build a slice explicitly.

Reply:
The other funcs could do the same if you move the cfgs to the last parameter.